core/src/main/java/greencity/service/impl/UserServiceImpl.java

```diff
@@ -224,15 +258,25 @@ public User update(UserUpdateDto dto, String email) {
     @Transactional
     @Override
     public List<UserGoalResponseDto> getUserGoals(Long userId, String language) {
-        List<UserGoalResponseDto> userGoalResponseDto = userGoalRepo
+        List<UserGoalResponseDto> userGoalResponseDtos = userGoalRepo
             .findAllByUserId(userId)
             .stream()
             .map(userGoal -> modelMapper.map(userGoal, UserGoalResponseDto.class))
             .collect(Collectors.toList());
-        if (userGoalResponseDto.isEmpty()) {
+        for (UserGoalResponseDto el1 : userGoalResponseDtos) {
+            if (el1.getText() == null) {
+                if (goalTranslationRepo.findByUserIdAndLanguageAndUserGoalId(userId, language, el1.getId()) != null) {
+                    el1.setText(goalTranslationRepo.findByUserIdAndLanguageAndUserGoalId(userId, language, el1.getId())
+                        .getText());
+                } else {
+                    el1.setText(customGoalRepo.findCustomGoalsForUserIdAndUserGoalId(el1.getId(), userId).getText());
+                }
+            }
+        }
```

Comment:
Please, try to refactor this code, it has low readability. At least move it to private method or maybe several methods.
Try to use Stream API here - this also could help.
You can write smth like userGoalResponseDtos.stream()
                                                .filter(el -> el.getText() == null)....